dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/persistence/tlvformat/TLVBlock.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.core.internal.persistence.tlvformat
+
+import com.datadog.android.api.InternalLogger
+import java.nio.ByteBuffer
+import java.util.Locale
+
+internal class TLVBlock(
+    val type: TLVBlockType,
+    val data: ByteArray,
+    val internalLogger: InternalLogger
+) {
+    @Suppress("ReturnCount")
+    internal fun serialize(maxEntrySize: Int = MAXIMUM_DATA_SIZE_MB): ByteArray? {
+        if (data.isEmpty()) return null
+
+        val typeFieldSize = Short.SIZE_BYTES
+        val dataLengthFieldSize = Int.SIZE_BYTES
+        val dataFieldSize = data.size
+
+        val entrySize = typeFieldSize + dataLengthFieldSize + dataFieldSize
+
+        if (entrySize > maxEntrySize) {
+            logEntrySizeExceededError(maxEntrySize)
+            return null
+        }
+
+        val tlvTypeAsShort = type.rawValue.toShort()
```

Comment:
minor nit: this is potentially lossy conversion because of the `UShort` vs `Short`. But ofc with our usage we will never hit it.